devito/types/basic.py

```diff
@@ -1247,6 +1248,19 @@ def _mem_mapped(self):
     def _mem_host(self):
         return self._space == 'host'
 
+    @cached_property
+    def _signature(self):
+        """
+        The signature of an AbstractFunction is the set of fields that
+        makes it "compatible" with another AbstractFunction. The fact that
```

Comment:
Might be worth defining "compatible" here

Reply:
that's encoded in the first part of the sentence

Reply:
In what sense? The current docstring doesn't help me understand what it means for one `AbstractFunction` to be compatible with another in this context

Reply:
I will try to improve it in my next PR